src/server/linux.rs

```diff
@@ -426,7 +426,22 @@ pub fn get_instance<'x>(method: &'x dyn Method, name: &str)
     -> anyhow::Result<InstanceRef<'x>>
 {
     let dir = data_dir()?.join(name);
-    if dir.exists() {
+    let name_os = dir.file_name().ok_or_else(|| {
+        InstanceNotFound(
+            anyhow::anyhow!("Invalid path: '{}'", dir.display())
+        )
+    })?;
+    if !dir.exists() {
+        Err(InstanceNotFound(
+            anyhow::anyhow!("Directory '{}' does not exist", dir.display())
+        ).into())
+    } else if unix::instance_name_from_os_str(name_os).is_none() {
```

Comment:
I'm not sure why this is needed? `name` == `name_os` here. And it's a contract that this should be a valid name here. We check for that either in CLI parser, or in directory listing.

If you think there is a bug somewhere, you can probably put an assertion. And assertion has to be before `dir.join()` otherwise this can be very interesting security issue.

Reply:
Shoot, forgot we have name here ... this check is to make sure _localdev is not treated as a local instance, therefore `instance unlink _localdev` could run properly.

Reply:
If there already is an instance named `_localdev` then users should be warned on unlink and should be able to destroy it. There is no reason to filter out it here.

Reply:
Per discussion, we won't reserve `_localdev` in this PR.